src/ontology/solvers/backend/OntologyConstraintSolver.java

```diff
@@ -95,9 +95,16 @@ protected InferenceSolution graphSolve(ConstraintGraph constraintGraph,
         } catch (InterruptedException | ExecutionException e) {
             e.printStackTrace();
         }
+
         InferenceSolution mergedSolution = mergeSolution(inferenceSolutionMaps);
 
-        verifyMergedSolution(mergedSolution, constraints, qualHierarchy);
+        try {
+            verifyMergedSolution(mergedSolution, constraints, qualHierarchy, inferenceSolutionMaps);
+        } finally {
```

Comment:
because in `verifyMergedSolution()` it uses `ErrorReporter.errorAbort()` to report errors if it founds solutions that violates constraints. In that situation, using `finally` block will ensure we will still get statistic information of post-verification even if an exception thrown from `verifyMergedSolution()` when it founds violated solutions.